conda/install.py

```diff
@@ -598,6 +605,14 @@ def unlink(prefix, dist):
         with open(meta_path) as fi:
             meta = json.load(fi)
 
+        # TODO Refactor to if not pkg.is_writable
+        if on_win and not can_open_all_files_in_prefix(prefix, meta['files']):
+            # TODO Should eventually return rather than exiting
+            sys.exit(
```

Comment:
@ilanschnell pointed out that if we exit here, we should do this before we do any unlinking, or else we could leave the environment in a broken state (e.g., if we unlink Python but don't link the new version).